pkg/util/ipvs/kernel_check.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipvs
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	utilsexec "k8s.io/utils/exec"
+
+	"github.com/golang/glog"
+)
+
+const (
+	// IPVSProxyMode is match set up cluster with ipvs proxy model
+	IPVSProxyMode = "ipvs"
+)
+
+// Sets of IPVS required kernel modules.
+var ipvsModules = []string{
+	"ip_vs",
+	"ip_vs_rr",
+	"ip_vs_wrr",
+	"ip_vs_sh",
+	"nf_conntrack_ipv4",
+}
+
+// KernelModuleIPVSRequiredCheck tests IPVS required kernel modules.
+type KernelModuleIPVSRequiredCheck struct{}
+
+// Name returns label for KernelModuleIPVSRequiredCheck
+func (KernelModuleIPVSRequiredCheck) Name() string {
+	return "KernelModuleIPVSRequired"
+}
+
+// LinuxKernelHandler implements KernelHandler interface.
```

Comment:
What is the `KernelHandler` interface?

Reply:
For `handle.executor.Command()`to read related file, we follow `ipvs`'s format.